OpenUpSA/wazimap-ng-ui
# aborting pending api requests when navigating

## Description
**_issue_** : 
when the user navigates between geographies too quickly, in the data mapper sometimes the categories of the wrong geography is shown

**_fix_** : 
when the user navigates to a new geography, we now abort the pending api requests to these endpoints : 
    * themes_count
    * all_details
    * indicator
    * children-indicators

this exposed some bugs we hadn't realized we had, like : 
   * rich data panel was intermittently showing the data of the wrong geography 
   * theme location summary in the rich data panel was failing to load

these bugs are fixed too

## Related Issue
https://wazimap.atlassian.net/browse/WNCM-434

## How is it tested automatically?
added test steps into `__tests__/gui/data_mapper.feature`

## How should a reviewer test it locally
* open developer console in SANEF profile
* as soon as you type something in the developer console you might get an alert message - ignore this
* run this script 
`window.location = '#geo:WC';
setTimeout(function(){
    window.location = '#geo:ZA';
}, 100)`
* this command will navigate to WC and after 100 ms back to ZA
* confirm that in data mapper and rich data panel, we display the ZA data correctly
* repeat this changing the delay. e.g change the `100` at the end to `500`, `1000`...

## Screenshots


## Changelog

### Added

### Updated
* `src/js/versions/version_controller.js` to properly reset when navigating between geographies
* `src/js/api.js` to abort pending requests when navigating
* `src/js/error-notifier.js` to prevent alert message when a pending api call is aborted

### Removed


## Checklist


### Pull Request


### Code Quality


### Testing
